processing/src/main/java/io/druid/query/Query.java

```diff
@@ -70,8 +70,18 @@
 
   String getType();
 
+  /**
+   * @deprecated use {@link QueryPlus#run(QuerySegmentWalker, Map)} instead. This method could be removed in the next
```

Comment:
In backwards incompatible patch, `Query.run(walker, context)` is removed and `Query.getRunner(walker, context)` could be added. Then `QueryPlus.run(walker, context)` is implemented as `this.query.getRunner(walker, context).run(this, context)`.